politeiawww/cmd/politeiavoter/politeiavoter.go

```diff
@@ -570,70 +573,99 @@ func (c *ctx) eligibleVotes(vrr *v1.VoteResultsReply, ctres *pb.CommittedTickets
 	return eligible, nil
 }
 
-func (c *ctx) _inventory() (*v1.ActiveVoteReply, error) {
-	responseBody, err := c.makeRequest(http.MethodGet, v1.RouteActiveVote, nil)
+func (c *ctx) _inventory(page uint32) (*tkv1.InventoryReply, error) {
+	i := tkv1.Inventory{
+		Page:   page,
+		Status: tkv1.VoteStatusStarted,
+	}
+	responseBody, err := c.makeRequest(http.MethodPost,
+		tkv1.APIRoute+tkv1.RouteInventory, i)
 	if err != nil {
 		return nil, err
 	}
 
-	var ar v1.ActiveVoteReply
+	var ar tkv1.InventoryReply
 	err = json.Unmarshal(responseBody, &ar)
 	if err != nil {
-		return nil, fmt.Errorf("Could not unmarshal ActiveVoteReply: %v",
+		return nil, fmt.Errorf("Could not unmarshal InventoryReply: %v",
 			err)
 	}
 
 	return &ar, nil
 }
 
 func (c *ctx) inventory() error {
-	i, err := c._inventory()
+	// Get server public key to verify replies.
+	version, err := c.getVersion()
 	if err != nil {
 		return err
 	}
-
-	// Get latest block
-	ar, err := c.wallet.Accounts(c.wctx, &pb.AccountsRequest{})
-	if err != nil {
-		return err
+	serverPubKey := version.PubKey
+	// Inventory route is paginated, therefore we keep fetching
+	// until we receive a patch with number of records smaller than the
+	// ticketvote's declared page size.
+	page := uint32(1)
+	var tokens []string
+	for {
+		ir, err := c._inventory(page)
+		if err != nil {
+			return err
+		}
+		tokens = append(tokens,
+			ir.Vetted[tkv1.VoteStatuses[tkv1.VoteStatusStarted]]...)
+		if uint32(len(tokens)) < tkv1.InventoryPageSize {
+			break
+		}
+		page++
 	}
-	latestBlock := ar.CurrentBlockHeight
-	//fmt.Printf("Current block: %v\n", latestBlock)
 
-	for _, v := range i.Votes {
-		// Make sure we have a CensorshipRecord
-		if v.Proposal.CensorshipRecord.Token == "" {
-			// This should not happen
-			log.Debugf("skipping empty CensorshipRecord")
-			continue
+	for _, t := range tokens {
+		// Get vote details.
+		d := tkv1.Details{
+			Token: t,
+		}
+		responseBody, err := c.makeRequest(http.MethodPost,
+			tkv1.APIRoute+tkv1.RouteDetails, d)
+		if err != nil {
+			return err
 		}
 
-		// Make sure we have valid vote bits
-		if v.StartVote.Vote.Token == "" || v.StartVote.Vote.Mask == 0 ||
```

Comment:
This validation isn't something that is checked by `client.VoteDetailsVerify()`, but probably should be.  Can you add this in?